### PR TITLE
 Add several functions to work with refGene exon sequences

### PR DESCRIPTION
**Pull request by alumi**
#### Summary
This PR adds several utility functions to `varity.ref-gene` module.

- refGene record => region
- refGene record => sequence of exons
- refGene record => base sequence

#### Tests

- `lein check` 🆗
- `lein test :all` 🆗

#### Notes
- Upgraded dependencies
- Replaced `varity.util/revcomp-bases` with more efficient implementation in cljam 0.6.0

### CODE REVIEW

File: src/varity/ref_gene.clj [hunk 1]
```diff
@@ -137,6 +138,73 @@
...
+         (map-indexed
+          (fn [i [s e]]
+            {:exon-index (inc i), ;; 1-origin
+             :exon-count exon-count,
+             :chr chr,
+             :start s,
+             :end e,
+             :reverse? reverse?})))))
+
+(defn cds-exon-seq
```
Comment (totakke) [inline src/varity/ref_gene.clj]: I think `cds-seq` is correct name. `cds-exon-` is wordy because CDS does not include introns.

**Comment by alumi** [comment]
@totakke Done, thanks for your review! 😸

File: src/varity/ref_gene.clj [hunk 2]
```diff
@@ -137,6 +139,73 @@
...
+             (update :end min cds-end))))
+     (exon-seq gene))))
+
+(defn ^String read-exon-sequence
+  "Reads a base sequence of an `exon` from `seq-rdr`."
+  [seq-rdr {:keys [reverse?] :as exon}]
+  (cond-> (cseq/read-sequence seq-rdr exon)
+    reverse? util-seq/revcomp))
+
+(defn ^String read-gene-sequence
```
Comment (federkasten) [inline src/varity/ref_gene.clj]: The `gene` word contains more widely meanings. Thus I think you should use `transcript` here.

Additionally, could you add docstrings to clarify the function, such as `the sequence contains 5'-UTR, CDS, and 3'-UTR`.

**Comment by alumi** [comment]
@federkasten I have replaced 'gene' with 'transcript' or 'ref-gene-record'.

Also I've added a commit to fix \:strand values (please refer to https://github.com/chrovis/cljam/pull/143#discussion_r213180699)
I'll open another PR for fixing existing functions like https://github.com/chrovis/cljam/pull/144

Thank you.

**Comment by totakke** [comment]
Thank you! The additional commits also seem good, so I've just merged.

**Comment by alumi** [comment]
@totakke @federkasten Thank you for your helpful comments! 😹

**Pull request merged**